ui/app/helpers/utils/util.js

```diff
@@ -465,18 +465,30 @@ export function constructTxParams({
  * or throws an error in case of failure.
  */
 export async function jsonRpcRequest(rpcUrl, rpcMethod, rpcParams = []) {
+  let fetchUrl = rpcUrl
+  const headers = {
+    'Content-Type': 'application/json',
+  }
+  // Convert basic auth URL component to Authorization header
+  const authMatches = rpcUrl.match('(http[s]?)://(.+):(.+)@(.+)')
```

Comment:
There is a much safer way for us to parse URLs: `[URL](https://developer.mozilla.org/en-US/docs/Web/API/URL)`.

e.g.
```
const { username, password } = new URL(rpcUrl)
```

If the browser can do it for us, lets take advantage! There are a lot of edge cases that this regex might not account for.

Reply:
thanks @Gudahtt for the suggestion, just pushed an update to accommodate that